### PR TITLE
Add a code action to add missing union constructors

### LINKED ISSUE #476: Offer to create missing union constructor
If there is no definition for a union constructor and we know the type that it should be, offer to create a variant on that type. This would be most useful with the `Msg` type, when it is common to write out the name of the msg you need before creating it.

### PR DESCRIPTION
**Pull request by jmbockhorst**
Closes #476.

### CODE REVIEW

**Comment by razzeee** [comment]
Tests seem unhappy

**Comment by jmbockhorst** [comment]
Whoops, I changed the name at the last minute. Still not sure about union variant/type/constructor naming.

**Review by razzeee — CHANGES_REQUESTED** [review]
Just some minors, looks good otherwise

File: src/util/refactorEditUtils.ts [hunk 1]
```diff
@@ -486,6 +486,31 @@ export class RefactorEditUtils {
...
+      const spaces = getSpaces(
+        lastUnionVariant.previousSibling?.startPosition.column ??
+          lastUnionVariant.startPosition.column,
+      );
+      return TextEdit.insert(
+        Position.create(
+          lastUnionVariant.endPosition.row,
+          lastUnionVariant.endPosition.column,
+        ),
+        `\n${spaces}| ${name}${params.length > 0 ? " " : ""}${params
```
[the remark this answers is not in the document]
Comment (jmbockhorst) [reply src/util/refactorEditUtils.ts:507]: When `params === 0` there is still a `name`, so it will be line `| NewMsg`. Params is just the part after that.

**Review by razzeee — APPROVED** [review]

**Pull request merged**